test/envoye2e/env/setup.go

```diff
@@ -432,6 +445,69 @@ func (s *TestSetup) VerifyStats(expectedStats map[string]int, port uint16) {
 	s.t.Errorf("failed to find expected stats: %v", err)
 }
 
+// VerifyPrometheusStats verifies prometheus stats.
+func (s *TestSetup) VerifyPrometheusStats(expectedStats map[string]int, port uint16) {
+	s.t.Helper()
+
+	check := func(respBody string) error {
+		for eStatsName, eStatsValue := range expectedStats {
+			var parser expfmt.TextParser
+			reader := strings.NewReader(respBody)
+			mapMetric, err := parser.TextToMetricFamilies(reader)
```

Comment:
Move this out of the loop. Parsing is expensive.

Reply:
Done.. Thanks!

Reply:
Wait, this is not done, right? Still parsing inside the loop.

Reply:
looks like none of my changes got pushed...

Reply:
Fixed now